src/plugins/saved_objects_management/public/plugin.ts

```diff
@@ -65,12 +84,27 @@ export class SavedObjectsManagementPlugin
       category: FeatureCatalogueCategory.ADMIN,
     });
 
+    const actionSetup = this.actionService.setup();
+
+    registerManagementSection({
+      core,
+      serviceRegistry: this.serviceRegistry,
+      sections: management.sections,
+    });
+
+    // depends on `getStartServices`, should not be awaited
+    registerServices(this.serviceRegistry, core.getStartServices);
```

Comment:
Yet another example of the `getStartServices` limitations. I'm supposed to register the services during `setup`, however some services requires `start` APIs. So I register them here, but do not await the registering promise (that waits for `getStartServices`) to resolve in `setup`. 

I could have performed the registration in the plugin's `start` instead, but that kinda go against the `setup/start` responsibilities (even if the registration is effectively performed after `start` resolves).

Reply:
Yea, it's the sync `ISavedObjectsManagementServiceRegistry` registry issue, with items requiring `start` services to be constructed. The root cause being indeed the `discover.savedSearches.createLoader` implementation.

These registrations and the whole service/loader registry will hopefully be removed (or at least adapted) in the follow-up, which is why I'm with this temporary 'workaround' for now.

I could move this `registerServices` call to the plugin's `start` method instead, as it's effectively doing the same thing. Tell me if we think this is better than running a non-awaited promise from `setup`.

Reply:
Nice to see platform team to also create a NP plugin and see limitations of `getStartServices`.